src/pages/index.js

```diff
@@ -1,50 +1,178 @@
+}
+
+
+const loadCardsCallback = (data) => {
+  const renderCards = new Section({items: data, renderer: (cardItem) => {
+    const cardElement = createNewCard(cardItem);
+
+    if(cardItem.isLiked){
+      cardElement.querySelector('.card__love-icon').classList.add('card__love-icon_background_black');
+    } else{
+      cardElement.querySelector('.card__love-icon').classList.remove('card__love-icon_background_black');
+    }
+
+    renderCards.appendItem(cardElement);
+  }}, cardContainerClass);
+
+  renderCards.renderItems();
+}
 
-  renderCards.prependItem(cardElement);
+
+const editProfileCallback = (values) => {
+  userInfo.setUserInfo(values);
+}
+
+
+const addNewCardCallback = (values) => {
+  const titleValue = values.name;
+  const urlValue = values.link;
+
+  const cardElement = createNewCard({name: titleValue, link: urlValue, _id: values._id});
+  document.querySelector('.elements').prepend(cardElement);
   formValidatorCardModal.disableStateButton();
+}
+
+
+//API CALLS
+const api = new Api(options);
+
+Promise.all([api.loadUserInfo(), api.loadCards()])
+  .then((data) => {
+    loadUserInfoCallback(data[0]);
+    loadCardsCallback(data[1]);
+  })
+  .catch(err => console.log(err))
+
+
+const addPlacePopup = createPopupWithForm('#add-card-modal', (inputValues) => {
+  renderLoading(true, cardModal);
+  api.addNewCard(inputValues)
+    .then(values => {
+      addNewCardCallback(values);
+      addPlacePopup.close();
+    })
+    .catch(err => console.log(err))
+    .finally(() => {
+      renderLoading(false, cardModal);
+    })
+
 });
 
 const userInfoPopup = createPopupWithForm('#profile-modal', (inputValues) => {
-  userInfo.setUserInfo(inputValues);
+  renderLoading(true, profileModal)
+  api.editProfile(inputValues, profileModal, userInfoPopup)
+    .then(values => {
+      editProfileCallback(values);
+      userInfoPopup.close();
+    })
+    .catch(err => console.log(err))
+    .finally(() => {
+      renderLoading(false, profileModal);
+    });
 });
 
-const renderCards = new Section({items: initialCards, renderer: (cardItem) => {
-  const cardElement = createNewCard(cardItem);
+const avatarPopup = createPopupWithForm('#avatar-modal', (inputValues) => {
+  renderLoading(true, avatarModal);
+  api.updateProfilePicture(inputValues)
+    .then(values => {
+      document.querySelector('.profile__image').src = values.avatar;
+      avatarPopup.close();
+    })
+    .catch(err => console.log(err))
+    .finally(() => {
+      renderLoading(false, avatarModal);
+    })
+})
 
-  renderCards.appendItem(cardElement);
-}}, cardContainerClass);
-
-renderCards.renderItems();
 
+const deleteCardCallback = (data, cardElement) => {
+  const deleteCardPopup = createPopupWithConfirmation('#delete-card-modal', () => {
+    api.deleteCard(data._id)
+    .then(() => {
+      deleteCardPopup.close();
+      cardElement.remove();
+    })
+    .catch(err => console.log(err))
+  });
 
+  deleteCardPopup.setEventListeners();
```

Comment:
`  deleteCardPopup.setEventListeners();
`

Please, move this into the file body

`.setEventListeners()` should be removed from `openings` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**